src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -875,25 +876,18 @@ public ImmutableArray<T> RemoveAll(Predicate<T> match)
             Requires.NotNull(match, "match");
 
             if (self.IsEmpty)
-            {
-                return new ImmutableArray<T>(self.array);
-            }
+                return self;
 
-            List<int> removeIndexes = null;
+            List<int> removeIndexes = new List<int>();
```

Comment:
:exclamation: I'm not convinced this will be faster than the previous lazy initialization.